internal/subtitles/client.go

```
@@ -0,0 +1,249 @@
+		return nil, fmt.Errorf("subtitles: subtitle file exceeds %d MiB limit", maxDownloadBytes>>20)
+	}
+	return data, nil
+}
+
+// maxDownloadBytes caps a subtitle download; a real .srt is kilobytes.
+const maxDownloadBytes = 10 << 20
+
+// validateLinkURL enforces the download-link policy (absolute https, or
+// loopback http for tests) on the initial link and on every redirect target.
+func validateLinkURL(u *url.URL) error {
+	if u == nil || u.Host == "" || !(u.Scheme == "https" || (u.Scheme == "http" && isLoopbackHost(u.Hostname()))) {
```

golangci-lint (2.12.2)

[error] QF1001: could apply De Morgan's law

(staticcheck)